src/launchoptions.rs

```diff
@@ -0,0 +1,211 @@
+use std::{
+    fs,
+    fs::{File, OpenOptions},
+    io::{Read, Write},
+    path::PathBuf,
+};
+
+use anyhow::{Context, Result};
+use regex::{Match, Regex};
+use steamid_ng::SteamID;
+use substring::Substring;
+use tracing::Level;
+
+use crate::gamefinder::locate_steam_launch_configs;
+
+pub const TF2_GAME_ID: &str = "440";
+pub const CONDEBUG: &str = "-condebug";
+pub const CONCLEARLOG: &str = "-conclearlog";
+pub const USERCON: &str = "-usercon";
+pub const G15: &str = "-g15";
+pub fn get_required_args_vec() -> Vec<String> {
+    let args: &[&str; 5] = &[TF2_GAME_ID, CONDEBUG, CONCLEARLOG, USERCON, G15];
+    let mut buf: Vec<String> = Vec::new();
+    args.iter().for_each(|str| buf.insert(0, str.to_string()));
+    buf
+}
+
+pub struct LaunchOptionsV2 {
+    local_config: PathBuf,
+    target_app: String,
+    launch_args_regex: Regex,
+    app_data: Option<String>,
+    new_app_data: Option<String>,
+    required_options: Vec<String>,
+}
+
+/// TODO: Implement a 'write_new_app_data_to_file' fn to write the updated launch options back to the
+/// file specified by 'local_config' struct element.
+/// TODO: Add more tracing to this to be nicer to the user.
+impl LaunchOptionsV2 {
+    pub fn new(user: SteamID, target_app: String) -> Result<LaunchOptionsV2, anyhow::Error> {
+        let span = tracing::span!(Level::INFO, "LaunchOptions");
+        let _enter = span.enter();
+
+        let required_opts = get_required_args_vec();
+        let config_path: PathBuf = locate_steam_launch_configs(user)
+            .context("Failed to find localconfig.vdf in Steam dir.")?;
+        let mut data: Vec<u8> = Vec::new();
+        tracing::debug!("Found Steam local app config.");
+
+        if let Ok(mut f) = File::open(&config_path) {
+            f.read_to_end(&mut data)
+                .context(format!("Failed to read {:?}", config_path))?;
+        }
+        let binding = &String::from_utf8_lossy(&data);
+
+        let apps_regex_raw = r#"\t{4}"apps"([\s\S]+)\t{5}}"#;
+        let apps_regex =
+            Regex::new(apps_regex_raw).context("Regex construction failed, good work.")?;
```

Comment:
Since this is a static regex, compilation failure should probably panic so we know there's an error in the code. We don't want to propagate this error, so it should be unwrapped with `expect`.